src/sentry/sentry_metrics/multiprocess.py

```diff
@@ -388,16 +416,51 @@ def process_messages(
             new_payload_value = deepcopy(parsed_payload_value)
 
             metric_name = parsed_payload_value["name"]
+            org_id = parsed_payload_value["org_id"]
             tags = parsed_payload_value.get("tags", {})
 
+            new_tags: MutableMapping[int, int] = {}
+
+            sentry_sdk.set_extra("orginal_tags", tags)
+
+            for k, v in tags.items():
+                tag_mappings: MutableMapping[str, int] = {}
+                tag_mappings.update(mapping.get(org_id, {}))
+                tag_mappings.update(mapping.get(0, {}))
```

Comment:
It looks like `tag_mappings` is independent of `k` and `v`. So you could put it outside the `for` loop, and then also use it on line 451 to get the `metric_id`.

Reply:
Why are you executing this logic for each tag?
Basically this means that, for each tag you instantiate a mapping and copy, one by one, all the mappings from the `mapping` dictionary. 
That dictionary is always going to be the same, so please move it outside of the loop.